source/core/oxseoencoder.php

```diff
@@ -1232,13 +1232,9 @@ public function fetchSeoUrl($sStdUrl, $iLanguage = null)
 
         $sShopId = $this->getConfig()->getShopId();
 
-        $sQ = "SELECT `oxseourl`, `oxlang` FROM `oxseo` WHERE `oxstdurl` = " . $oDb->quote($sStdUrl) . " AND `oxlang` = '$iLanguage' AND `oxshopid` = '$sShopId' LIMIT 1";
-
-        $oDb = oxDb::getDb(oxDb::FETCH_MODE_ASSOC);
-        $oRs = $oDb->select($sQ);
-
-        if (!$oRs->EOF) {
-            $sSeoUrl = $oRs->fields['oxseourl'];
+        $sQ = "SELECT `oxseourl` FROM `oxseo` WHERE `oxstdurl` = " . $oDb->quote($sStdUrl) . " AND `oxlang` = '$iLanguage' AND `oxshopid` = '$sShopId' LIMIT 1";
+        if($sOxseourl = $oDb->getOne($sQ)) {
```

Comment:
be aware that empty string "" and "0" will than not be valid values for oxseourl anymore .